### PR TITLE
[IMP] l10n_hk: added FPS QR code payment for invoice

### PR DESCRIPTION
**Pull request by tong-odoo**
- Add FPS QR code payment by following the government document of FPS QR Code Standard
- https://www.hkma.gov.hk/media/eng/doc/key-functions/financial-infrastructure/infrastructure/retail-payment-initiatives/Common_QR_Code_Specification.pdf

task-3042940

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account_qr_code_emv/tests/test_emv_qr.py
Comment (rigr-odoo) [inline addons/account_qr_code_emv/tests/test_emv_qr.py:29]: Command.create is clearer than 0, 0

**Review by rigr-odoo — COMMENTED** [review]
`.pot` files are not really needed in master :)

Besides that, this seems good to me, thx!

File: addons/account_qr_code_emv/i18n/account_qr_code_emv.pot
Comment (rigr-odoo) [inline addons/account_qr_code_emv/i18n/account_qr_code_emv.pot:1]: Not needed :p

**Comment by rigr-odoo** [comment]
LGTM thx, it has been forwarded to a colleague for the final review :)

**Comment by tong-odoo** [comment]
@rigr-odoo May I ask is it possible to add this in 16.3?

File: addons/account_qr_code_emv/tests/test_emv_qr.py
Comment (jco-odoo) [inline addons/account_qr_code_emv/tests/test_emv_qr.py:1]: Better to remove this test as it will only fail with the right things installed...

**Review by jco-odoo — COMMENTED** [review]
The little module seems a bit overkill, but we should be able to use it for many more countries apparently.

**Comment by tong-odoo** [comment]
> The little module seems a bit overkill, but we should be able to use it for many more countries apparently.

Yes, the main idea is for other countries as well

**Comment by jco-odoo** [comment]
@tong-odoo If in the manifest we describe it is just a bridge module (it does not work on its own) and we remove the test in that module, I think we can go forward with it if @rigr-odoo agrees.

**Comment by tong-odoo** [comment]
> @tong-odoo If in the manifest we describe it is just a bridge module (it does not work on its own) and we remove the test in that module, I think we can go forward with it if @rigr-odoo agrees.

Thanks for the review! I updated the commit. Please take a look.

**Comment by jco-odoo** [comment]
@tong-odoo Could you still add a bit more info in the first commit message: [ADD] account_qr_code_emv: support EMV codes

And then some explanation maybe about how it will be used in more countries?

**Comment by tong-odoo** [comment]
> @tong-odoo Could you still add a bit more info in the first commit message: [ADD] account_qr_code_emv: support EMV codes
> 
> And then some explanation maybe about how it will be used in more countries?

@jco-odoo Updated the commit message. Thanks!

File: addons/l10n_hk/models/res_bank.py
Comment (oco-odoo) [inline addons/l10n_hk/models/res_bank.py:15]: Just call it FPS ID. It'll be shorter and cleaner ; even when it contains a mail, this mail IS the FPS id.

Shouldn't those two fields be declared in account_qr_code_emv ? Is there that much flexibility in what can go into that field in other countries needing it ?

**Review by oco-odoo — CHANGES_REQUESTED** [review]

**Review by vin-odoo — APPROVED** [review]
LGTM at this point. Lets keep in mind that we could make the CURRENCY_MAPPING into a regular field on res.currency if at some point we need it again for some other systems.

@oco-odoo @jco-odoo @rigr-odoo Let's merge this today if it's ok with you :)

**Comment by jco-odoo** [comment]
@tong-odoo @vin-odoo I suggest we merge already.  But I don't feel the comments of @oco-odoo being totally addressed however about the label of the field:  we could e.g. put the type and identifier on one line in the view so we don't have to bother about the actual label.

**Comment by jco-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by jco-odoo** [comment]
@robodoo r-

File: addons/l10n_hk/models/res_bank.py
Comment (oco-odoo) [reply addons/l10n_hk/models/res_bank.py:15]: Hmm, ok. We'll see if the field needs to be moved in the parent module when we add other modules depending on this type of QR code. 

For the label, I really don't agree: it's too long and it's not clear what should go into it because of this A/B/C style. Plus, if "FPS ID" is not clear enough, then the field should not be called l10n_hk_fps_identifier. 

Anyway, with the imminent freeze, I'll try some changes myself, so that we can move on faster :)

Comment (oco-odoo) [reply addons/l10n_hk/models/res_bank.py]: Discussed with jco: let's just call that field "FPS Identifer" ; should be clear enough, and we can change that based on feedbacks later on if experience shows there is confusion among the users between this and "FPS ID" selection value.

**Comment by oco-odoo** [comment]
@robodoo r+

**Comment by Xavier-Do** [comment]
@robodoo r-
Looks like it is the cause of https://runbot.odoo.com/runbot/batch/1048780/build/36189714

**Comment by vin-odoo** [comment]
@robodoo r+

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**
